File: fire-client-java/src/main/java/es/gob/clavefirma/client/InvalidTransactionException.java
/* Copyright (C) 2017 [Gobierno de Espana]
 * This file is part of FIRe.
 * FIRe is free software; you can redistribute it and/or modify it under the terms of:
 *   - the GNU General Public License as published by the Free Software Foundation;
 *     either version 2 of the License, or (at your option) any later version.
 *   - or The European Software License; either version 1.1 or (at your option) any later version.
 * Date: 08/09/2017
 * You may contact the copyright holder at: soporte.afirma@correo.gob.es
 */
package es.gob.clavefirma.client;

/**
 * Excepci&oacute;n que se&ntilde;ala que se ha intentado acceder a los datos de una transaccion
 * inexistente o caducada.
 * @author Carlos Gamuci
 */
public class InvalidTransactionException extends HttpOperationException {

	/** Serial Id. */
	private static final long serialVersionUID = 3011445782702735307L;

	/** Construye la excepci&oacute;n. */
	public InvalidTransactionException() {
		super();
	}

	/** Construye la excepci&oacute;n indicando un mensaje.
	 * @param msg Mensaje descriptivo del error. */
	public InvalidTransactionException(final String msg) {
		super(msg);
	}

	/** Construye la excepci&oacute;n indicando la causa.
	 * @param cause Causa de la excepci&oacute;n. */
	public InvalidTransactionException(final Throwable cause) {
		super(cause);
	}

	/** Construye la excepci&oacute;n con su descripci&oacute;n y la causa.
	 * @param msg Mensaje de error.
	 * @param cause Causa de la excepci&oacute;n. */
	public InvalidTransactionException(final String msg, final Throwable cause) {
		super(msg, cause);
	}
}

File: fire-client-dotnet/FIRe/FIRe/InvalidTransactionException.cs
/* Copyright (C) 2017 [Gobierno de Espana]
 * This file is part of FIRe.
 * FIRe is free software; you can redistribute it and/or modify it under the terms of:
 *   - the GNU General Public License as published by the Free Software Foundation;
 *     either version 2 of the License, or (at your option) any later version.
 *   - or The European Software License; either version 1.1 or (at your option) any later version.
 * Date: 08/09/2017
 * You may contact the copyright holder at: soporte.afirma@correo.gob.es
 */

using System;

namespace FIRe
{
    /// <summary>
    /// Indica cuando se intenta operar sobre una transacción no existente o ya caducada.
    /// </summary>
    class InvalidTransactionException : HttpOperationException
    {
        /// <summary>
        /// Se crea la excepción.
        /// </summary>
        public InvalidTransactionException() : base()
        {
        }

        /// <summary>
        /// Se crea la excepción.
        /// </summary>
        /// <param name="code">Código del error.</param>
        /// <param name="msg">Descripcion del error.</param>
        public InvalidTransactionException(int code, string msg) : base(code, msg)
        {
        }

        /// <summary>
        /// Se crea la excepción.
        /// </summary>
        /// <param name="msg">Descripcion del error.</param>
        /// <param name="e">Causa del error</param>
        public InvalidTransactionException(string msg, Exception e) : base(msg, e)
        {
        }
    }
}
